lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/Faucet.hs

```diff
@@ -0,0 +1,271 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cardano.Wallet.Jormungandr.Faucet
+    ( initFaucet
+
+    -- * Internal
+    , genFaucets
+    ) where
+
+import Prelude hiding
+    ( appendFile )
+
+import Cardano.Wallet.Api.Types
+    ( encodeAddress )
+import Cardano.Wallet.Jormungandr.Binary
+    ( MkFragment (..), TxWitnessTag (..), fragmentId, putFragment )
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( AccountingStyle (..)
+    , HardDerivation (..)
+    , NetworkDiscriminant (..)
+    , PaymentAddress (..)
+    , SomeMnemonic (..)
+    , publicKey
+    )
+import Cardano.Wallet.Primitive.AddressDerivation.Icarus
+    ( generateKeyFromSeed )
+import Cardano.Wallet.Primitive.Fee
+    ( FeePolicy (..) )
+import Cardano.Wallet.Primitive.Mnemonic
+    ( Mnemonic, entropyToMnemonic, genEntropy, mnemonicToText )
+import Cardano.Wallet.Primitive.Types
+    ( Address (..), Coin (..), Hash (..), TxIn (..), TxOut (..) )
+import Cardano.Wallet.Unsafe
+    ( unsafeFromHex )
+import Control.Concurrent.MVar
+    ( newMVar )
+import Control.Monad
+    ( forM_, replicateM )
+import Data.ByteString
+    ( ByteString )
+import Data.Text
+    ( Text )
+import GHC.TypeLits
+    ( KnownNat )
+import System.FilePath
+    ( FilePath, (</>) )
+import System.IO.Temp
+    ( withSystemTempDirectory )
+import Test.Integration.Faucet
+    ( Faucet (..), icaMnemonics, rndMnemonics, seqMnemonics )
+import Test.Integration.Jcli
+    ( argHex, argInt, getBlock0H, jcli, jcli_, sinkAddress )
+
+import qualified Codec.Binary.Bech32 as Bech32
+import qualified Codec.Binary.Bech32.TH as Bech32
+import qualified Data.ByteString as BS
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+import qualified Data.Text.IO as TIO
+
+-- | Initialize a bunch of faucet wallets and make them available for the
+-- integration tests scenarios.
+initFaucet :: FeePolicy -> IO Faucet
+initFaucet policy = Faucet
+    <$> newMVar seqMnemonics
+    <*> newMVar icaMnemonics
+    <*> newMVar rndMnemonics
+    <*> newMVar (mkTxBuilder policy <$> externalAddresses)
+
+-- | Prepare externally signed Tx for Jormungandr
+mkTxBuilder
+    :: FeePolicy
+    -> (TxIn, String)
+    -> (Address, Coin)
+    -> IO ByteString
+mkTxBuilder (LinearFee cst coeff _) (TxIn inpTx inpIx, key) (addr, Coin amt) =
+    withSystemTempDirectory "cardano-wallet-jormungandr" $ \d -> do
+        let txFile = d </> "trans.tx"
+        let witnessFile = d </> "witness"
+        let keyFile = d </> "key.prv"
```

Comment:
wonder if we should not have everywhere consistent policy like you used in `teardownConfig` when persisting to filesystem

Reply:
What do you mean :thinking: ? (about the file extensions I picked for the node: node.config, node.key, node.cert, node.topology etc... ?)

Reply:
I meant that for Byron.Integration you used `teardownConfig` which removed tmp files unless `NO_CLEANUP` is set. And wonder if cleaning "policy" should not be the same here in Jormungandr. Or we should also use `NO_CLEANUP` here to have the same cleaning policy of tmp dir.

Reply:
It is the same in Jörmungandr actually but you have to compare the same elements (i.e. the configuration generation) :wink: 

--> https://github.com/input-output-hk/cardano-wallet/blob/master/lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/Launch.hs#L64-L74

In the `Faucet` however, those temporary files are merely used as buffers because it is trickier to provide inputs on stdin when invoking `jcli` from Haskell.